### PR TITLE
Implement socket_handle

### LINKED ISSUE #31: Implement socket resource and connect algorithm

### PR DESCRIPTION
**Pull request by maikel**
closes #31

### CODE REVIEW

File: source/sio/io_uring/socket_handle.hpp [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    };
+
+    template <class Receiver>
+    struct operation_base
+      : stoppable_op_base<Receiver>
+      , submission {
+      operation_base(
+        int fd,
+        ip::endpoint peer_endpoint,
+        exec::io_uring_context& context_,
```
Comment (emmett2020) [inline source/sio/io_uring/socket_handle.hpp]: may remove underscore

File: source/sio/io_uring/socket_handle.hpp [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        Receiver rcvr) noexcept
+        : stoppable_op_base<Receiver>{context_, static_cast<Receiver&&>(rcvr)}
+        , submission{fd, peer_endpoint} {
```
Comment (emmett2020) [inline source/sio/io_uring/socket_handle.hpp:103]: should we make a rvalue-reference there

File: source/sio/io_uring/socket_handle.hpp [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    connect_::sender connect(async::connect_t, ip::endpoint peer_endpoint) noexcept {
+      return {this->context_, peer_endpoint, fd_};
+    }
+
+    ip::endpoint local_endpoint() const;
+    ip::endpoint remote_endpoint() const;
+  };
+
+  template <class Protocol>
+  struct socket_resouce {
```
Comment (emmett2020) [inline source/sio/io_uring/socket_handle.hpp]: may rename to 'resource'

**Review by emmett2020 — APPROVED** [review]
It is awesome. I only commented some misspelling :)

File: source/sio/io_uring/socket_handle.hpp (hunk 3 above)
Comment (maikel) [reply source/sio/io_uring/socket_handle.hpp]: good catch, thank you

File: source/sio/io_uring/socket_handle.hpp (hunk 2 above)
Comment (maikel) [reply source/sio/io_uring/socket_handle.hpp:103]: Isn't peer endpoint trivialy copy constructible? How would that be for e.g. windows?

Comment (emmett2020) [reply source/sio/io_uring/socket_handle.hpp:103]: Oh, yes. I read through the implementation. It really is trivially copy constructible.

**Pull request merged**